### PR TITLE
Added Resource Request objects to components missing them

### LINKED ISSUE #650: add resource requests to hypershift operator install and review to ensure all components have requests
hypershift operator currently when getting installed does not set any resource requests. That should be done. In addition any other components should be scanned to see if they don't have any

### PR DESCRIPTION
**Pull request by jzielny**
**What this PR does / why we need it**:
Added Resource Request objects to components missing them (hypershift-operator, machine-config-server, openshift-oauth-apiserver, control-plane-operator)
**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #650 

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/install/assets/hypershift_operator.go [hunk 1]
```diff
@@ -237,6 +238,12 @@ func (o HyperShiftOperatorDeployment) Build() *appsv1.Deployment {
 									Protocol:      corev1.ProtocolTCP,
 								},
 							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceMemory: resource.MustParse("157286400"),
```
Comment (enxebre) [inline cmd/install/assets/hypershift_operator.go:243]: what's this value coming from?

Comment (jzielny) [reply cmd/install/assets/hypershift_operator.go]: They were the corresponding values for 150Mi and 10m as integers but I've normalized them below to make them more readable

**Comment by csrwng** [comment]
@jzielny looks like you need to run `make update`

**Comment by jzielny** [comment]
Updated the saas_template using your recommendation @csrwng

**Pull request merged**